pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -423,24 +423,90 @@ func (rq *ResourceQuotaController) Sync(discoveryFunc NamespacedResourcesFunc, p
 			return
 		}
 
-		// Something has changed, so track the new state and perform a sync.
-		klog.V(2).Infof("syncing resource quota controller with updated resources from discovery: %v", newResources)
-		oldResources = newResources
-
 		// Ensure workers are paused to avoid processing events before informers
 		// have resynced.
 		rq.workerLock.Lock()
 		defer rq.workerLock.Unlock()
 
-		// Perform the monitor resync and wait for controllers to report cache sync.
-		if err := rq.resyncMonitors(newResources); err != nil {
-			utilruntime.HandleError(fmt.Errorf("failed to sync resource monitors: %v", err))
-			return
+		// Once we get here, we should not unpause workers until we've successfully synced
```

Comment:
so if a resource (say from aggregated apiserver) somehow is discoverable, but not syncable, that will block resourcequota from working for all other resources, even though that particular problematic resource is not used anywhere for quota.

Reply:
that's true... this was copied from the GC controller, where the consequences of incomplete information are far worse (incorrect object deletion).

I think I can simplify this to just timeout the WaitForCacheSynced line and add logic to the quota evaluators to error if their listers are consulted before their backing informers are fully synced.